controllers/pod_controller.go

```diff
@@ -60,7 +60,36 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *PodReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+	// Index endpoints by UID. Later when we reconcile the Pod this will make it easy to find the associated endpoints
+	// and auto populate DNS names.
+	err := mgr.GetFieldIndexer().IndexField(ctx, &corev1.Endpoints{}, reconciler.EndpointUID, func(rawObj client.Object) []string {
+		log := log.FromContext(ctx)
```

Comment:
log is only used whn error, can you move into if?

Reply:
Done